homeassistant/components/rfxtrx.py

```diff
@@ -173,13 +177,22 @@ def handle_receive(event):
     device = config[DOMAIN][ATTR_DEVICE]
     debug = config[DOMAIN][ATTR_DEBUG]
     dummy_connection = config[DOMAIN][ATTR_DUMMY]
+    recv_modes = config[DOMAIN][ATTR_RECV_MODES]
+
+    if recv_modes is not None:
```

Comment:
Expose this into a function they trow a vol.Invalid(). And use that on config check. We should not initialize this if the config is not valid.

Reply:
I don't quite understand what you mean here. 

In my first version of the code, I did try to  [validate the configuration directly](https://github.com/home-assistant/home-assistant/pull/10203/files/4f0095a2aacb13bf4367cc2b45fa63def5e9864b#diff-54e1ac0eeac5c94c45a700374d717628), but as @Danielhiversen pointed out, the code needs to access the pyRFXtrx lib dependency, so it does not work on a fresh install because the pyRFXTRX lib is imported after the config validation occurs.

That's why I've changed the code to perform the validation in the setup() method.

Can you give me more hints on what to do then? Thanks!

Reply:
@pvizeli : As @ypollart  says: We can not import the pyRFXtrx library in config validation. So another option is to copy the information from the library to this class. But it is not good to have the information duplicated.

In the future, I think we should find a way to import libraries in config validation.

Reply:
yeah, I see